charms/istio-pilot/lib/charms/istio_pilot/v0/istio_gateway_info.py

```diff
@@ -50,15 +88,41 @@ def some_event_function():
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 1
+LIBPATCH = 2
 
-
-RELATION_NAME = "gateway-info"
-INTERFACE_NAME = "istio-gateway-info"
+# Default relation and interface names. If changed, consistency must be kept
+# across the provider and requirer.
+DEFAULT_RELATION_NAME = "gateway-info"
+DEFAULT_INTERFACE_NAME = "istio-gateway-info"
 
 logger = logging.getLogger(__name__)
 
 
+def relation_preflight_checks(
+    model: Model, relation_name: str, requirer: bool = False
+) -> None:
+    """Series of checks for ensuring the relation is properly established.
+    Args:
+        model (Model): the juju model as seen by the provider/requirer unit
+        relation_name (str): the name of the relation
+        requirer (bool, optional): set to True if the check is run in the requirer class
+    Raises:
+        GatewayRelationMissingError when there is no relation between two units
+        GatewayRelationDataMissingError when there is no data in the data bag
+    """
+    # Do nothing if the unit is not leader
+    if not model.unit.is_leader():
+        logger.info("This unit is not leader, no action will be taken.")
+        raise RuntimeError("Unit is not leader")
+
+    relation = model.get_relation(relation_name)
+    if not relation:
+        raise GatewayRelationMissingError()
+
+    if requirer and not relation.data:
+        raise GatewayRelationDataMissingError("No data found in relation data bag.")
```

Comment:
should this assert that data exists, or that *valid* data exists (eg check for the expected data keys, like done [here](https://github.com/canonical/istio-operators/pull/237/files#diff-85507bb21a0ce73a75d60021f5ae8970194bd2e968a5e1fc0b954a41acae3adeR186-R193).  I think those checks could be pulled into here